File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_map<int, bool> isPresent;

        for(auto num : nums){
            if(isPresent[num]){
                return true;
            }
            
            isPresent[num] = true;
        }

        return false;
    }
};

File: 0217-contains-duplicate/0217-contains-duplicate.cs
public class Solution {
    public bool ContainsDuplicate(int[] nums) {
        var mp = new Dictionary<int, int>();

        foreach(var num in nums){
            if(mp.ContainsKey(num)){
                return true;
            }

            mp[num] = 1;
        }

        return false;
    }
}